test/models/translations_test.rb

```diff
@@ -42,19 +42,30 @@ def simple_type(x)
       x.is_a?(Proc)
   end
 
+  # Is the HTML string acceptable?  It needs to NOT have common mistakes,
+  # *and* have only the permitted HTML tags & attributes.
   def acceptable_html_string(text)
-    # In the longer term we should probably only accept lower case, but
-    # HTML doesn't care about case, so don't worry about it for now.
-    return false if %r{<[^A-Za-z\/]}.match?(text)
-    return false if %r{<\/[^A-Za-z]}.match?(text)
+    # First, detect common mistakes.
+
+    # Require HTML tags to start in a lowercase Latin letter.
+    # This is in part a regression test; it prevents </a> where "a"
+    # is the Cyrillic letter instead of the Latin letter.
+    # HTML doesn't care about upper vs. lower case,
+    # but it's better to be consistent, and there's a minor
+    # compression advantage as described here:
+    # http://www.websiteoptimization.com/speed/tweak/lowercase/
+    return false if %r{<[^a-z\/]}.match?(text)
+    return false if %r{<\/[^a-z]}.match?(text)
+    return false if text.include?('href = ')
```

Comment:
The speed up is not even measurable. I tried. But it is the right thing to do, because if a string of text is not HTML, the other tests should not be applied,  and this is an easy and reasonable test to check for HTMLness.